cpp/csp/engine/Struct.h

```
@@ -75,12 +105,14 @@ class StructField
 protected:
 
     StructField( CspTypePtr type, const std::string & fieldname,
-                 size_t size, size_t alignment );
+                 size_t size, size_t alignment, bool optional );
 
     void setIsSet( Struct * s ) const
     {
         uint8_t * m = reinterpret_cast<uint8_t *>( s ) + m_maskOffset;
         (*m) |= m_maskBitMask;
+
+        clearIsNone( s ); // no-op if not an optional field
```

no-op in that it defaults to 0 offset / 0 mask? 
I assume so, in which case we might want to comment where its initialized that we depend on that for the no-op here